enhancements/sig-architecture/34-work-executor-group/README.md

```diff
@@ -0,0 +1,147 @@
+# Work Executor Group
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management-io/website/)
+
+## Summary
+
+The proposal adds a new field `executorGroupName` to the spec of `ManifestWork` 
+api, to separate the execution of a set of `ManifestWork` resources to the 
+dedicated work agents. The default/legacy work agent will be the default
+executor of the `ManifestWork` resource if its `executorGroupName` is not
+specified.
+
+## Motivation
+
+### Controlling availability (blast-radius) of the work agents
+
+Currently, for each of the managed clusters we will deploy the work agents
+in a single deployment resource which is sharing the availability via "leader-
+election", so there's merely one instance of work agent taking effect at a time.
+Under this pattern, we can't break down the availability of the work agent into
+finer units so that:
+
+- During the upgrade, we can progressively reduce the impact from the agent's 
+  down-time.
+- By grouping the agents according to their use-cases, we can control the 
+  blast-radius so that the unavailability will not propogate between the groups.
+
+### Separate the identity and its held permission for the work agents
+
+There can be different actors using the work api to distribute resource from 
+the hub cluster. Currently the work agent has the cluster-admin access to the
+managed cluster which means that every hub users with write access to the
+work resources will be able to dispatch any resources to the managed cluster.
+By making different groups of work agents mount different service accounts and
+granting different scopes of permission to those service accounts, we will be
+able to further define the expected purpose/scope of the work agent groups.
+
+### Goals
+
+- `ManifestWork` resource be able to exclusively assign the execution to an 
+  expected group of work agents.
+
+### Non-Goals
+
+- This proposal doesn't cover the authorization of "which hub users can assign
+  work resources to which executor group".
+  
+## Proposal
+
+### API Changes
+
+This proposal adds a new field `executorGroupName` to the `ManifestWork` api, 
+which is defaulted to (the string) "default" if absent. Meanwhile, upon starting
+a work agent we can set its executor group via an optional flag:
+
+> `--executor-group=<group name>`: If the work agent is running with a non-empty
+> executor group, it will only manage those manifest-work resources that assigned
+> the same group name. By default, an empty group means the agent is the default
+> executor.
+
+```yaml
+apiVersion: work.open-cluster-management.io/v1
+kind: ManifestWork
+metadata: ...
+spec:
+  executor:
+    group: default|...
```

Comment:
Will there be restrictions about who can use certain executor groups?  

if not, what is the value of 
> By making different groups of work agents mount different service accounts and granting different scopes of permission to those service accounts, we will be able to further define the expected purpose/scope of the work agent groups.

it becomes a benefit of organization on the spokes, but not a restriction of power from hub?  That's ok with me, but I'd like to be sure I understand it.  Perhaps someone could write custom rules restricting usage in their hub.


If yes, this appears to complicate the handling of any abstraction on top of manifestwork.  I don't think I'd recommend restricting the ability to set this field out of the box.

Reply:
comparing to the relationship between the original service-account and the pods, the kube-apiserver doesnt authorize whether a hub user can mount a service-account when creating pods. but they can opt-in that authorization by developing a custom admission webhook. that's similar for the proposed api change in the work api, the work agent isnt able to authorize that until we plumb a validating admission webhook to the hub. i think the alpha phase wont cover the implementation of that webhook, but we can expect that in the beta. wdyt?